inspirehep/ext/jinja_filters/record.py

```diff
@@ -355,6 +355,48 @@ def epoch_to_year_format(date):
 
     @app.template_filter()
     def construct_date_format(date):
-        starting_date = time.strftime('%Y-%m-%d', time.gmtime(int(date) / 1000.))
+        starting_date = time.strftime(
+            '%Y-%m-%d', time.gmtime(int(date) / 1000.))
         ending_date = time.strftime('%Y-12-31', time.gmtime(int(date) / 1000.))
         return starting_date + '->' + ending_date
+
+    @app.template_filter()
+    def author_urls(l, separator):
+        result = []
+        for name in l:
+            if 'name' in name:
+                url = '<a href="/search?ln=en&amp;cc=HepNames&amp;p=name:'\
+                      + name['name'] + '&amp;of=hd">' + name['name'] + '</a>'
+                result.append(url)
+        return separator.join(result)
+
+    @app.template_filter()
+    def ads_links(record):
+        import re
+        lastname = ''
+        firstnames = ''
+        initial = ''
+        link = ''
+        re_last_first = re.compile(
```

Comment:
So I think the compilation is only advantageous when you run the expression many times, but here it is *maybe* run once. So perhaps either (a) move the compilations to a global in the file. E.g. like here: https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/migrator/tasks.py#L60, or (b) do not compile them but use them directly when needed, like `re.search`.